pkg/agent/downloader.go

```diff
@@ -20,19 +20,22 @@ import (
 	"encoding/hex"
 	"encoding/json"
 	"fmt"
-	"github.com/kubeflow/kfserving/pkg/agent/storage"
-	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
-	"github.com/pkg/errors"
-	"go.uber.org/zap"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"regexp"
 	"strings"
+	"sync"
+
+	"github.com/kubeflow/kfserving/pkg/agent/storage"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
 type Downloader struct {
-	ModelDir  string
+	ModelDir string
+	sync.Mutex
```

Comment:
I actually prefer unexported type(`mu sync.Mutex`) here, embedded mutex struct is useful when you have lock/unlock as public API, but here I think lock is only used internally without exposing.

Reply:
I see your point and have updated.